### ISSUE TITLE
model->findFirstBy... doesn't work with columnMap() in Oracle 

### ISSUE THREAD

**Issue by skaosYan** [phalcon/cphalcon#10490]
Hi,

Phalcon 2.0.2, Oracle 12c.

I have set columnMap() for my model.
Then, I invoke find on my model like this:
Users::findFirstByEmail('test@test.com');

In SQL debug.log I see:

[Tue, 09 Jun 15 02:58:38 +0300][INFO] SELECT COUNT(*) "numrows" FROM (SELECT * FROM (SELECT Z1.*, ROWNUM PHALCON_RN FROM (SELECT USERS.USERID, USERS.NAME, USERS.EMAIL, USERS.PASSWORD, USERS.MUSTCHANGEPASSWORD, USERS.PROFILESID, USERS.SEX, USERS.PHONE, USERS.BIRTHDATE FROM USERS WHERE USERS.EMAIL = :0) Z1 WHERE ROWNUM <= 1))

But the call itself results in error:

Column 'PHALCON_RN' doesn't make part of the column map<br>#0 [internal function]: Phalcon\Mvc\Model::cloneResultMap(Object(Vokuro\Models\Users), Array, Array, 0, false)<br />
#1 [internal function]: Phalcon\Mvc\Model\Resultset\Simple-&gt;current()<br />
#2 [internal function]: Phalcon\Mvc\Model\Resultset-&gt;getFirst()<br />
#3 [internal function]: Phalcon\Mvc\Model\Query-&gt;execute(Array, NULL)<br />
#4 [internal function]: Phalcon\Mvc\Model::findFirst(Array)<br />
#5 /home/var/www/html/v1/app/controllers/UsersController.php(93): Phalcon\Mvc\Model::__callStatic('findFirstByEmai...', Array)<br />
#6 /home/var/www/html/v1/app/controllers/UsersController.php(93): Vokuro\Models\Users::findFirstByEmail('test@test.ru')<br />
#7 [internal function]: Vokuro\Controllers\UsersController-&gt;createAction()<br />
#8 [internal function]: Phalcon\Dispatcher-&gt;dispatch()<br />
#9 /home/var/www/html/v1/public/index.php(41): Phalcon\Mvc\Application-&gt;handle()<br />

How this bug can be fixed?

**Comment by dreamsxin** [comment]
Please submit model code?

**Comment by skaosYan** [comment]
I have fixed it by adding 	    'PHALCON_RN' => 'phalcon_rn', into the columnMap(). But this should be done att core level.


<?php
namespace Vokuro\Models;

use Phalcon\Mvc\Model;
use Phalcon\Mvc\Model\Validator\Uniqueness;
use Phalcon\Mvc\Model\MetaData;
use Phalcon\Db\Column;


/**
 * Vokuro\Models\Users
 * All the users registered in the application
 */
class Users extends Model
{
    public $userid;
    public $name;
    public $email;
    public $password;
    public $phone;
    public $sex;
    public $birthdate;
    public $mustChangePassword;
    public $profilesId;

    public function getSequenceName()
    {
        return 'USERS_SEQ';
    }

    public function afterCreate()
    {
//	$userid = $this->getDI()->get('db')->lastInsertId($this->getSequenceName());
    } 

    public function getSource()
    {
        return "USERS";
    }

    public function columnMap()
    {
        return array(
            'USERID' => 'userid',
            'NAME' => 'name',
            'PHONE' => 'phone',
            'EMAIL' => 'email',
            'PASSWORD' => 'password',
            'MUSTCHANGEPASSWORD' => 'mustChangePassword',
            'PROFILESID' => 'profilesId',
            'SEX' => 'sex',
            'BIRTHDATE' => 'birthdate',
	    'PHALCON_RN' => 'phalcon_rn',
        );
    }
   
    public function initialize() {
      $this->skipAttributesOnCreate(
        array('USERID')
      );
    }



}

**Comment by dreamsxin** [comment]
Fields should case sensitive.

**Comment by skaosYan** [comment]
Sorry, I do not understand how to apply your answer, would you elaborate?

**Comment by dreamsxin** [comment]
Column name is `phalcon_rn`, Use  uppercase `PHALCON_RN` must be map.

**Comment by andresgutierrez** [comment]
This is fixed in the 2.0.x branch (Phalcon 2.0.4), you have to change a global ORM setting to allow unexpected columns:

```php
\Phalcon\Mvc\Model::setup(['ignoreUnknownColumns' => true]);
```

**Comment by skaosYan** [comment]
Cool, thank you
